Makefile

```diff
@@ -224,4 +224,14 @@ validate-no-changes: clean compile docs
 
 .PHONY: npm-publish
 npm-publish:
+	# this set is taken from esbuild's process https://github.com/evanw/esbuild/blob/master/Makefile#L330
+	@npm --version > /dev/null || (echo "The 'npm' command must be in your path to publish" && false)
+	@echo "Checking for uncommitted/untracked changes..." && test -z "`git status --porcelain | grep -vE 'M (CHANGELOG\.md|version\.txt)'`" || \
```

Comment:
Why the references to CHANGE.md and version.txt?

(Also we don't have a version.txt file.)

Reply:
oh oops i need to adjust this for our case thanks for catch

Reply:
ok i fixed this -- it basically validates no changes to anything besides changelog.md and package.json (eventually this is what a Github action will run to change the changelog from `Unreleased` to `<new version>` and package.json, make that commit and then publish)

Reply:
So for now is the flow that you manually bump the version in CHANGELOG.md and package.json, publish the new package, and then merge your changes? Or should you merge the changes first?

Reply:
hmm yeah maybe ill remove the omitting changelog and package.json for now since prior to github action you need to manually merge it in and then run npm-publish